paper/paper.md

```diff
@@ -0,0 +1,135 @@
+In the previous example, this could be done using the following code:
+
+```python
+converted = rdata.read_rda(
+    "dataset.rda",
+    constructor_dict={"factor": factor_constructor},
+)
+```
+
+When the default conversion routine is being executed, if an object belonging to an S3 or S4 class is found, the appropriate constructor will be called passing to it the partially constructed object.
+If no constructor is available for that class, a warning will be emitted and the constructor of the most immediate parent class available will be called.
+If there are no constructors for any of the parent classes, the basic underlying Python object will be left without transformation.
+
+By default, a dictionary named `DEFAULT_CLASS_MAP` is passed to `convert()` including constructors for commonly used classes, such as `data.frame`, `ordered` or the aforementioned `factor`.
+In case the user desires different conversions for basic R objects, it would be enough to create a subclass of the `Converter` class.
+Several utility functions, such as the routines `convert_char()` and `convert_list()`, are exposed by the conversion module in order for users to be able to reuse them for that purpose.
+
+# Ongoing work
+
+To broaden the utility of the `rdata` library for data processing pipelines with steps in both R and Python, we are currently extending the library with the capability to write compatible Python objects to Rda and Rds files.
+As an example, such a pipeline is present in the Hmsc-HPC code [@rahman+_2024_hmschpc].
+The continuous development of this code has also been driving the ongoing work on the writing functionality of the `rdata` library.
+The writing of Rda and Rds files is implemented as a two-step process similar to reading: first, the Python object is converted to the tree-like intermediate representation used in parsing, and then this intermediate representation is written to a file of the chosen format.
+Currently, the writing functionality supporting common types is available in the development branch of the `rdata` library.
+
+# Acknowledgements
+
+This work has received funding
+from the Spanish Ministry of Education and Innovation, projects PID2019-106827GB-I00 / AEI / 10.13039/501100011033 and PID2019-109387GB-I00,
+from an FPU grant (Formación de Profesorado Universitario) from the Spanish Ministry of Science, Innovation and Universities(MICIU) with reference FPU18/00047,
+and from the European Union's Horizon Europe research and innovation programme under grant agreement No 101057437 (BioDT project, [https://doi.org/10.3030/101057437](https://doi.org/10.3030/101057437)).
+Views and opinions expressed are those of the author(s) only and do not necessarily reflect those of the European Union or the European Commission. Neither the European Union nor the European Commission can be held responsible for them.
+
```

Comment:
Adding missing acknowledgement.

```suggestion
The authors gratefully acknowledge the use of the computational facilities of Centro de Computación Científica (CCC) at Universidad Autónoma de Madrid.

```

Reply:
```suggestion
The authors gratefully acknowledge the use of the computational facilities provided by Centro de Computación Científica (CCC) at Universidad Autónoma de Madrid and by CSC – IT Center for Science, Finland.

```

I'd also like to add similar acknowledgement to [our facilities](https://docs.csc.fi/support/faq/how-to-cite-csc/). I also changed "of" to "provided by" to make the sentence structure work with both acknowledgements.